components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -951,6 +955,79 @@ public static SessionContext getSessionContextFromCache(String key) {
+                boolean isSessionExpired = sessionContextCache.isSessionExpired(cacheKey, cacheEntry);
+                if (isSessionExpired) {
+                    triggerSessionExpireEvent(request, context, sessionContext);
+                    if (log.isDebugEnabled()) {
+                        log.debug("A SESSION_EXPIRE event was fired for the expired session found corresponding " +
+                                "to the key: " + cacheKey.getContextId());
+                    }
+                    return null;
+                }
+            }
+        }
+        return sessionContext;
+    }
+
+    /**
+     * Trigger SESSION_EXPIRE event on session expiry due to a session idle timeout or a remember me session time out.
+     *
+     * @param request        HttpServletRequest.
+     * @param context        Authentication context.
+     * @param sessionContext Session context.
+     * @throws FrameworkException Error in triggering the session expiry event.
+     */
+    private static void triggerSessionExpireEvent(HttpServletRequest request, AuthenticationContext context,
+                                                  SessionContext sessionContext) throws FrameworkException {
+
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        if (sessionContext != null) {
+            Object authenticatedUserObj = sessionContext.getProperty(FrameworkConstants.AUTHENTICATED_USER);
+            if (authenticatedUserObj instanceof AuthenticatedUser) {
+                authenticatedUser = (AuthenticatedUser) authenticatedUserObj;
+            }
+            context.setSubject(authenticatedUser);
+
+            IdentityEventService eventService = FrameworkServiceDataHolder.getInstance().getIdentityEventService();
+            try {
+                Map<String, Object> eventProperties = new HashMap<>();
+                eventProperties.put(IdentityEventConstants.EventProperty.REQUEST, request);
+                eventProperties.put(IdentityEventConstants.EventProperty.CONTEXT, context);
+                eventProperties.put(IdentityEventConstants.EventProperty.SESSION_CONTEXT, sessionContext);
+                Map<String, Object> paramMap = new HashMap<>();
+                paramMap.put(FrameworkConstants.AnalyticsAttributes.USER, authenticatedUser);
+                paramMap.put(FrameworkConstants.AnalyticsAttributes.SESSION_ID, context.getSessionIdentifier());
+                Map<String, Object> unmodifiableParamMap = Collections.unmodifiableMap(paramMap);
+                eventProperties.put(IdentityEventConstants.EventProperty.PARAMS, unmodifiableParamMap);
+
+                Event event = new Event(IdentityEventConstants.EventName.SESSION_EXPIRE.name(), eventProperties);
+                eventService.handleEvent(event);
+            } catch (IdentityEventException e) {
+                throw new FrameworkException("Error in triggering the session expire event.", e);
```

Comment:
Let's have some context information. Like the sessionId, user etc.

Reply:
Fixed with cf0c1c45baba0a400fc721abd4eb7e9833acbc4d